plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1063,6 +1064,250 @@ public void testCreatePartitionedTableWithQuotedIdentifierCasing(String columnNa
+                "  'a_boolean', " +
+                "  'an_integer', " +
+                "  'a_bigint', " +
+                "  'a_real', " +
+                "  'a_double', " +
+                "  'a_short_decimal', " +
+                "  'a_long_decimal', " +
+                "  'a_varchar', " +
+                "  'a_varbinary', " +
+                "  'a_date', " +
+                "  'a_time', " +
+                "  'a_timestamp', " +
+                "  'a_timestamptz', " +
+                "  'a_uuid'" +
+                "  ]" +
+                ")");
+        String values = "(" +
+                "true, " +
+                "1, " +
+                "BIGINT '2', " +
+                "REAL '3.0', " +
+                "DOUBLE '4.0', " +
+                "DECIMAL '5.00', " +
+                "DECIMAL '6.00', " +
+                "'seven', " +
+                "X'88888888', " +
+                "DATE '2022-09-09', " +
+                "TIME '10:10:10.000000', " +
+                "TIMESTAMP '2022-11-11 11:11:11.000000', " +
+                "TIMESTAMP '2022-11-11 11:11:11.000000 UTC', " +
+                "UUID '12121212-1212-1212-1212-121212121212', " +
+                "ROW(13, 'thirteen'), " +
+                "ARRAY['four', 'teen'], " +
+                "MAP(ARRAY[15], ARRAY['fifteen']))";
+        String highValues = "(" +
+                "true, " +
+                "999999999, " +
+                "BIGINT '999999999', " +
+                "REAL '999.999', " +
+                "DOUBLE '999.999', " +
+                "DECIMAL '999.99', " +
+                "DECIMAL '6.00', " +
+                "'zzzzzzzzzzzzzz', " +
+                "X'FFFFFFFF', " +
+                "DATE '2099-12-31', " +
+                "TIME '23:59:59.999999', " +
+                "TIMESTAMP '2099-12-31 23:59:59.000000', " +
+                "TIMESTAMP '2099-12-31 23:59:59.000000 UTC', " +
+                "UUID 'FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF', " +
+                "ROW(999, 'zzzzzzzz'), " +
+                "ARRAY['zzzz', 'zzzz'], " +
+                "MAP(ARRAY[999], ARRAY['zzzz']))";
+        String lowValues = "(" +
+                "false, " +
+                "0, " +
+                "BIGINT '0', " +
+                "REAL '0', " +
+                "DOUBLE '0', " +
+                "DECIMAL '0', " +
+                "DECIMAL '0', " +
+                "'', " +
+                "X'00000000', " +
+                "DATE '2000-01-01', " +
+                "TIME '00:00:00.000000', " +
+                "TIMESTAMP '2000-01-01 00:00:00.000000', " +
+                "TIMESTAMP '2000-01-01 00:00:00.000000 UTC', " +
+                "UUID '00000000-0000-0000-0000-000000000000', " +
+                "ROW(0, ''), " +
+                "ARRAY['', ''], " +
+                "MAP(ARRAY[0], ARRAY['']))";
+
+        assertUpdate("INSERT INTO " + tableName + " VALUES " + values + ", " + highValues + ", " + lowValues, 3);
+        dropTable(tableName);
+    }
+
```

Comment:
```
trino:default> CREATE TABLE my_table2 (
            ->         c1 integer,
            ->         c2 date,
            ->         c3 double
            ->     )
            ->     WITH (
            ->         format = 'PARQUET',
            ->         partitioning = ARRAY['c1', 'c2'],
            ->         sorted_by = ARRAY['c4']
            ->     );
Query 20221222_101514_00001_qebin failed: Cannot find field 'c4' in struct: struct<1: c1: optional int, 2: c2: optional date, 3: c3: optional double>
org.apache.iceberg.exceptions.ValidationException: Cannot find field 'c4' in struct: struct<1: c1: optional int, 2: c2: optional date, 3: c3: optional double>
```

```
trino:default> CREATE TABLE my_table2 (
            ->         c1 integer,
            ->         c2 date,
            ->         c3 double
            ->     )
            ->     WITH (
            ->         format = 'PARQUET',
            ->         partitioning = ARRAY['missingpartitionfield'] );
Query 20221222_101645_00003_qebin failed: Unable to parse partitioning value: Cannot find source column: missingpartitionfield
io.trino.spi.TrinoException: Unable to parse partitioning value: Cannot find source column: missingpartitionfield
```

The checks regarding the specified `partitioning` , `sorted_by` field are made only when passing the command to Iceberg framework even though they are actually semantic exceptions that may be caught at Trino level. See the following example where the catching of the exception happens at Trino level:

```
trino:default> CREATE TABLE my_table2 (
            ->         c1 integer,
            ->         c1 date
            ->     )
            ->     WITH (
            ->         format = 'PARQUET');
Query 20221222_102120_00014_qebin failed: line 3:9: Column name 'c1' specified more than once
io.trino.spi.TrinoException: line 3:9: Column name 'c1' specified more than once
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
	at io.trino.execution.CreateTableTask.internalExecute(CreateTableTask.java:170)
```

I see this issue rather as minor/ pre-existing in case of `partitioning` and can be definitely done as a follow-up.